app/scripts/views/Calendar.js

```diff
@@ -77,6 +82,55 @@ class Calendar extends Component {
     )
   }
 
+  renderMap() {
+    if (!this.props.listItems.events) {
+      return null
+    } else if (this.props.listItems.events.listItems[0]) {
+      const allEvents = this.props.listItems.events.listItems
+      const allVenues = []
+      allEvents.map(item => {
+        const filterVenues = allEvents.filter(i => ((item.placeId === i.placeId)))
+        const exists = allVenues.filter(x => {
+          return x.find(({ placeId }) => placeId === item.placeId)
+        })
+        if (exists.length === 0) {
+          allVenues.push(filterVenues)
+          return allVenues
+        } return null
```

Comment:
I'm not sure if I understand this block of code, can you walk me through what you want to do?

Also, some style nitpicks:

* Prefer less generic variable names (not `x` or `i` or `exists`)
* The `return null` is somehow confusingly placed as one does not see it so easily behind the bracket. I would give it its own new line.

Reply:
Ah yeh, vvvvvery messy, sorry. It's for filtering events by venue, so a list of events happening at each venue can be shown. However, I was coming into loads of problems because of null values being pushed into arrays and then throwing errors later. So this is basically my hacky solution to that problem.... I think I'm just missing some fundamental idea.... I am looking more into best practice for filtering so I think I'll find the correct solution along the way. But if you have pointers....

Reply:
No problem! Its a great feature :+1:

I'm wondering if it makes sense to take the events from the `infiniteList` store as it is paginated, which means you will only see the events in the map which are currently loaded in the paginated view.

Think the default is 50 for the first load right now? There will be events missing as soon as the festival contains more than this number and as long as the user didn't click to load more.

Reply:
Ah, gotcha. Where would be the best place to grab them from? I can't find another example of fetching a list of events. I'll refactor this whole section of code once I have a good plan.